ksql-engine/src/main/java/io/confluent/ksql/codegen/CodeGenRunner.java

```diff
@@ -129,7 +130,7 @@ protected Object visitFunctionCall(FunctionCall node, Object context) {
       String functionName = node.getName().getSuffix();
       KsqlFunction ksqlFunction = functionRegistry.getFunction(functionName);
       parameterMap.put(
-          node.getName().getSuffix(),
+          node.getName().getSuffix() + "_" + functionCounter++,
```

Comment:
It's important that the name generated by this Visitor match the name generated by SqlToJavaVisitor, which feels brittle to me. I think a better way of doing this might be to move building the parameter map into SqlToJavaVisitor so that it returns code and a spec of the expected parameters. What do you think?

Reply:
I agree, but I wanted to keep this change small. The work I've got coming up on quoted identifiers will see big changes in this area of the code, so I figured I'd learn the code now and make the bigger changes later.

Reply:
Sounds good to me.